src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/TypeBuilderImpl.cs

```diff
@@ -79,9 +81,101 @@ protected override void AddInterfaceImplementationCore([DynamicallyAccessedMembe
 
         [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2083:DynamicallyAccessedMembers", Justification = "Not sure how to handle")]
         [return: DynamicallyAccessedMembers((DynamicallyAccessedMemberTypes)(-1))]
-        protected override TypeInfo CreateTypeInfoCore() => this;
-        protected override ConstructorBuilder DefineConstructorCore(MethodAttributes attributes, CallingConventions callingConvention, Type[]? parameterTypes, Type[][]? requiredCustomModifiers, Type[][]? optionalCustomModifiers) => throw new NotImplementedException();
-        protected override ConstructorBuilder DefineDefaultConstructorCore(MethodAttributes attributes) => throw new NotImplementedException();
+        protected override TypeInfo CreateTypeInfoCore()
+        {
+            if (_isCreated)
+            {
+                return this;
+            }
+
+            // Create a public default constructor if this class has no constructor. Except the type is Interface, ValueType, Enum, or a static class.
+            // (TypeAttributes.Abstract | TypeAttributes.Sealed) determines if the type is static
+            if (_constructorDefinitions.Count == 0 && (_attributes & TypeAttributes.Interface) == 0 && !IsValueType &&
+                ((_attributes & (TypeAttributes.Abstract | TypeAttributes.Sealed)) != (TypeAttributes.Abstract | TypeAttributes.Sealed)))
+            {
+                DefineDefaultConstructor(MethodAttributes.Public);
+            }
+
+            _isCreated = true;
+            return this;
+        }
+
+        private void ThrowIfTheTypeCreated()
```

Comment:
```suggestion
        private void ThrowIfCreated()
```
It is the usual naming convention for these methods in the repo. (E.g. look for `ThrowIfDisposed` or `ThrowIfNull`)